frontend/src/app/services/user-route-access.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from "@angular/router";
+import { KeycloakAngularModule, KeycloakService } from "keycloak-angular";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserRouteAccessService implements CanActivate {
```

Comment:
You can remove `UserRouteAccessService`, because you already have `AuthGuard` which implements the same interface `CanActivate`. See comment in `AuthGuard`